runtime/src/main/java/org/corfudb/runtime/collections/Table.java

```diff
@@ -169,10 +159,13 @@ public Table(@Nonnull final TableParameters<K, V, M> tableParameters,
      * Delete a record mapped to the specified key.
      *
      * @param key Key.
-     * @return Previously stored Corfu Record.
+     * @return Previously stored Corfu Record. Throws NoSuchElementException if key is not found.
      */
     @Nullable
     CorfuRecord<V, M> deleteRecord(@Nonnull final K key) {
+        if (!corfuTable.containsKey(key)) {
+            throw new NoSuchElementException("deleteRecord failed as key "+key.toString()+" does not exist");
```

Comment:
will it be thrown to the listeners?  If so, will they need to resubscribe?

Reply:
no, this is a CRUD api, it is not part of streaming